### PR TITLE
Add redux toolkit to bookstore

### PR DESCRIPTION
**Pull request by shahzaman900**
### Add redux toolkit to bookstore

1. Install the redux toolkit
2. create store 
3. create booksSlice file
4. create categorySlice file 
5. add functionality

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const books = createSlice({
+  name: '',
+  initialState: [],
+  reducers: {
+    addBook: (state) => state,
+    removeBook: (state) => state,
```
Comment (OyePriscilla) [inline src/redux/books/booksSlice.js]: - [ ] The logic for adding and removing book(s) is missing from your implementation, kindly reference this [tutorial](https://medium.com/@ralph1786/adding-and-removing-items-from-redux-store-6d1303ed32c6). ℹ️ Every time object is added or remove via the reducer, twoactions are performed to update the array either to add or remove. 🙏 

Untitled

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const books = createSlice({
+  name: '',
+  initialState: [],
```
Comment (OyePriscilla) [inline src/redux/books/booksSlice.js]: - [ ] Kindly take not that for proper identification, the name is not supposed to be empty. ℹ️ Kindly give it a descriptive name, maybe `books`.

File: src/redux/store.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+const store = configureStore({
+  reducer: {},
+});
```
Comment (OyePriscilla) [inline src/redux/store.js]: Kindly import the two reducers you implemented in the redux directory for them to be accesible by the store; don't forget to configure them into the reducer.

```Example
import { configureStore } from "@reduxjs/toolkit"
import booksReducer from "./books/booksSlice"
import categoriesReducer from "./categories/categoriesSlice"

const store = configureStore({
  reducer: { books: booksReducer, categories: categoryReducer },
})
;
```

The code snippet is just an example.

File: src/redux/categories/categoriesSlice.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const category = createSlice({
+  name: '',
+  initialState: [],
+  reducers: {
+    checkStatus: () => 'under construction',
+  },
+});
```
Comment (OyePriscilla) [inline src/redux/categories/categoriesSlice.js:9]: - [ ] Kindly take not that for proper identification, the name is not supposed to be empty. ℹ️ Kindly give it a descriptive name, maybe `categories`. 

- [ ] The logic for the category state is missing as well, kindy implement it. 👍 

```example
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
```
```

![Uploading Untitled.png…]()

Comment (OyePriscilla) [inline src/redux/categories/categoriesSlice.js]: - [ ] Kindly take not that for proper identification, the name is not supposed to be empty. ℹ️ Kindly give it a descriptive name, maybe `categories`. 

- [ ] The logic for the category state is missing as well, kindy implement it. 👍 

```example
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
```
Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @sheezu900 👍 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Added Redux Toolkit ✔️
Created a directory that will contained all your Redux logic ✔️ 

## Required Changes ♻️

- [x] I coudn't access your app because of the error screenshot below 👇 , kindly import `store.js` correctly in `index.js` file. 
`import Store from './redux/store';
`
Untitled



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by divinecharlotte — APPROVED** [review]
# Hello @sheezu900     😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say  👍🏻 ✅ 
### To Highlight 💯


- Good implementation of the required changes👍 
- well-organized structure✔ 👍
 




## Congratulations! 🎉 



Cheers, and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**